lua/cmp/utils/async.lua

```diff
@@ -58,8 +61,8 @@ async.throttle = function(fn, timeout)
         time = vim.loop.now()
       end
 
-      self.running = true
       self.stop(false)
+      self.running = true
```

Comment:
When I changed to `self.running = false`, the issue is solved and the stutter is gone. (context: https://github.com/hrsh7th/nvim-cmp/pull/1610#issuecomment-1583286858)

Reply:
That's not a real fix unfortunately.
I did however fix another issue in the code just now. Can you try again using this PR?

Reply:
Can't get it to work anymore:
```
Error executing vim.schedule lua callback: .../magnus/.local/share/nvim/lazy/nvim-cmp/lua/cmp/core.lua:294: attempt to index field 'filter' (a nil value)
stack traceback:
        .../magnus/.local/share/nvim/lazy/nvim-cmp/lua/cmp/core.lua:294: in function 'complete'
        .../magnus/.local/share/nvim/lazy/nvim-cmp/lua/cmp/core.lua:170: in function 'autoindent'
        .../magnus/.local/share/nvim/lazy/nvim-cmp/lua/cmp/core.lua:162: in function 'on_change'
        .../magnus/.local/share/nvim/lazy/nvim-cmp/lua/cmp/init.lua:313: in function 'callback'
        .../.local/share/nvim/lazy/nvim-cmp/lua/cmp/utils/async.lua:164: in function <.../.local/share/nvim/lazy/nvim-cmp/lua/cmp/utils/async.lua:162>
```

Reply:
just fixed that, but something is very broken apart from that with my PR. Will check what's wrong here

Reply:
Should be fixed now. I wasnt using this branch myself. I thought I still had lazy configured to use the dev version :)

Can you test with my latest changes?

Reply:
I tried https://github.com/hrsh7th/nvim-cmp/pull/1611/commits/0d8de773e08aca4e3be665a0ee8895127a25e36f but still stutters. Using the correct config, deleted `lazy-lock.json` and `nvim-cmp` plugin itself and ran `Lazy sync`.
```lua     {
        "folke/nvim-cmp",
        branch = "filter_sync",
```

Reply:
Found the issue and should be fixed now. Would be great if you can confirm